### DISCUSSION TITLE
Is it possible to hide documents of private modules?

### DISCUSSION THREAD

**Discussion by toku-sa-n** [ocaml/dune#8489] [Q&A] [answered]
Hi. After `dune init project foo`, I created `lib/hoge.ml` and `lib/fuga.ml` and modified `lib/dune`:

```dune
(library
 (name foo)
 (public_name foo)
 (modules hoge fuga)
 (private_modules fuga))
```

Then, I ran `dune build @doc`. As below, `Fuga` is listed, but I want not to list it as it is a private module.
![24-08-2023_23 24 45_screenshot](https://github.com/ocaml/dune/assets/42336339/053fbd6e-75c8-46fc-bd30-3a73921e3832)

I tried `(modules Hoge Fuga)(private_modules Fuga)` (capitalizing the first letter), but it didn't change anything.

Is it possible to hide it? If so, how to do it?

The version of my `dune` is 3.10.0.

Thanks.

**Comment by Alizter** [comment, marked as answer]
Libraries in Dune are wrapped by default meaning that Dune will generate a `foo.ml` file which exports all the modules of the library. This is called the top-level module. The reason you see `Hoge` is because you are looking at `Foo` the top-level module that Dune created (with both modules). To fix this, you can make your own top-level module. Just create a file `foo.ml` with the following:
```
module Fuga = Fuga
```

Read more about it here https://dune.readthedocs.io/en/stable/dune-files.html#library

**Reply by Alizter** [reply]
On the other hand, we should probably make it so that `private_modules` are not included in the toplevel module. cc @rgrinberg

**Comment by rgrinberg** [comment]
@jonludlam what do you think about this one?
